src/x_connective.c

```diff
@@ -1519,30 +1519,38 @@ typedef struct vcommon
 {
     t_pd c_pd;
     int c_refcount;
-    t_float c_f;
+    t_atom c_atom;
 } t_vcommon;
 
 typedef struct _value
 {
     t_object x_obj;
     t_symbol *x_sym;
-    t_float *x_floatstar;
+    t_atom *x_atomstar;
 } t_value;
 
-    /* get a pointer to a named floating-point variable.  The variable
+    /* get a pointer to a named float variable.  The variable
     belongs to a "vcommon" object, which is created if necessary. */
 t_float *value_get(t_symbol *s)
+{
+    t_atom *a = value_getatom(s);
+    return &a->a_w.w_float;
+}
+
+    /* get a pointer to a named variable's underlying atom.  The variable
+    belongs to a "vcommon" object, which is created if necessary. */
+t_atom *value_getatom(t_symbol *s)
 {
     t_vcommon *c = (t_vcommon *)pd_findbyclass(s, vcommon_class);
     if (!c)
     {
         c = (t_vcommon *)pd_new(vcommon_class);
-        c->c_f = 0;
```

Comment:
properly initialize as a float atom, e.g. `SETFLOAT(&c->c_atom, 0);`

Reply:
My original implementation required *not* setting the type, but this should be fine now.